### DISCUSSION TITLE
Deprecate `flat` eventually

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#84] [Ideas]
When we eventually update our browser support to those with ES2019, we should deprecate `flat` and recommend `Array.prototype.flat` instead. It supports both deep and shallow flattening.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat

**Comment by MarlonPassos-git** [comment]
@aleclarson Have you ever thought about this type of deprecation?

I think we could do something like this.

Occasionally, we'll upload the modification to add @deplacate to the JDOC function and some more console.warn() warning about the deprecation and reason
and then in the next major, we'll remove the `flat` function from the code.

Or we could use this here to discuss a deprecation strategy for radashi

**Reply by Minhir** [reply]
I would expect deprecation warning from some framework or core library. For the utils library, marking deprecation in the JSDoc should be enough.
